StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/ViewControllers/UpdatePaymentMethodViewController.swift

```diff
@@ -218,25 +218,31 @@ final class UpdatePaymentMethodViewController: UIViewController {
         }
         view.isUserInteractionEnabled = false
         updateButton.update(state: .spinnerWithInteractionDisabled)
-
-        var analyticsParams: [String: Any] = [:]
-
-        if case .card(let paymentMethodCardParams) = updatePaymentMethodOptions {
-            analyticsParams["selected_card_brand"] = paymentMethodCardParams.networks?.preferred
-        }
-        if let setAsDefaultValue {
-            analyticsParams["set_as_default"] = setAsDefaultValue
-        }
         do {
             try await delegate.didUpdate(viewController: self, paymentMethod: configuration.paymentMethod)
-            STPAnalyticsClient.sharedClient.logPaymentSheetEvent(event: configuration.hostedSurface.analyticEvent(for: .updateCard),
-                                                                 params: analyticsParams)
-        } catch {
+            if case .card(let paymentMethodCardParams) = updatePaymentMethodOptions,
+               let selectedCardBrand = paymentMethodCardParams.networks?.preferred {
+                STPAnalyticsClient.sharedClient.logPaymentSheetEvent(event: configuration.hostedSurface.analyticEvent(for: .updateCardBrand),
+                                                                     params: ["selected_card_brand": selectedCardBrand])
+            }
+            if setAsDefaultValue ?? false {
+                STPAnalyticsClient.sharedClient.logPaymentSheetEvent(event: configuration.hostedSurface.analyticEvent(for: .setDefaultPaymentMethod),
+                                                                     params: ["payment_method_type": configuration.paymentMethod.type.identifier])
+            }
+        } catch let error as NSError {
             updateButton.update(state: .enabled)
             latestError = error
-            STPAnalyticsClient.sharedClient.logPaymentSheetEvent(event: configuration.hostedSurface.analyticEvent(for: .updateCardFailed),
-                                                                 error: error,
-                                                                 params: analyticsParams)
+            if case NSError.stp_cardBrandNotUpdatedError() = error,
+               case .card(let paymentMethodCardParams) = updatePaymentMethodOptions,
+               let selectedCardBrand = paymentMethodCardParams.networks?.preferred {
+                STPAnalyticsClient.sharedClient.logPaymentSheetEvent(event: configuration.hostedSurface.analyticEvent(for: .updateCardBrandFailed),
+                                                                     params: ["selected_card_brand": selectedCardBrand])
+            }
+            if case NSError.stp_defaultPaymentMethodNotUpdatedError() = error,
+               setAsDefaultValue ?? false {
```

Comment:
Seems like we should always log this error analytic regardless of the value of "setAsDefaultValue" if stp_defaultPaymentMethodNotUpdatedError is thrown.